### PR TITLE
fix type annotations and convention ii

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

i was reading [this type annotation PEP](https://peps.python.org/pep-0585/) and realized i've been getting type annotations wrong
so i ctrl f'd and tried to fix all of them

### CODE REVIEW

**Review by ryanchou-dev — APPROVED** [review]
⭐

File: content/4_Gold/LIS.mdx [hunk 1]
```diff
@@ -92,10 +92,7 @@ public static int findLis(int[] a) {
 <PySection>
 
 ```py
-from typing import List
-
-
-def find_lis(a: List[int]) -> int:
+def find_lis(a: list[int]) -> int:
```
Comment (bqi343) [inline content/4_Gold/LIS.mdx:95]: this doesn't compile in python 3.8 and lower (https://stackoverflow.com/questions/39458193/using-list-tuple-etc-from-typing-vs-directly-referring-type-as-list-tuple-etc) so maybe just leave it as is? note that CF currently uses python 3.8 and USACO uses python 3.6

Comment (SansPapyrus683) [reply content/4_Gold/LIS.mdx]: alr, everything should be back to using `typing` now

**Comment by SansPapyrus683** [comment]
OH NO FORGOT THIS EXISTED

**Review by envyaims — APPROVED** [review]
code seems fine

File: solutions/usaco-859.mdx [hunk 2]
```diff
@@ -28,46 +29,172 @@ int main() {
...
+		}
+
+		// sort by arrival time
+		Arrays.sort(cows, Comparator.comparingInt(c -> c.start));
+	
+		int time = 0;
+		int curr = 0;
+		int longestWait = 0;
+
+		// sorted by priority so that the highest seniority starts eating first
```
Comment (envyaims) [inline solutions/usaco-859.mdx:106]:
```suggestion
		// sorted so that the highest seniority starts eating first
```
idk

**Pull request merged**